lib/eslint/eslint-helpers.js

```diff
@@ -675,7 +675,7 @@ function processOptions({
     overrideConfig = null,
     overrideConfigFile = null,
     plugins = {},
-    reportUnusedDisableDirectives = null, // ← should be null by default because if it's a string then it overrides the 'reportUnusedDisableDirectives' setting in config files. And we cannot use `overrideConfig.reportUnusedDisableDirectives` instead because we cannot configure the `error` severity with that.
+    reportUnusedDisableDirectives = null, // ← should be null by default because if it's a string then it overrides the 'reportUnusedDisableDirectives' setting in config files. And we cannot use `overrideConfig.reportUnusedDisableDirectives` instead because we cannot configure the `error` severity with that. TODO: should anything change based on this comment?
```

Comment:
As for the first sentence, I think nothing should change.

The second sentence, I believe, explains why this option is needed. After this change, integrations and our CLI will be able to set this value in `overrideConfig`, so we might consider dropping this option.

Reply:
I'll look into dropping this option. It wouldn't be a breaking change, would it?

Reply:
I think it wouldn't be a breaking change at this point since it's in the `FlatESLint` class, and I'm in favor of removing constructor options that turned out to be redundant with config options, but would like @nzakas's opinion too.

Reply:
I agree with removing redundant options. I added a note to the RFC about this:

> Phase 2: Any breaking changes for flat config users (`warn` by default, remove redundant `reportUnusedDisableDirectives` option from `processOptions()`), suitable for a major release or potentially a minor release since the flat config is not considered stable yet.